services/controllerhost/replicaStatsMRUCache.go

```diff
@@ -77,47 +78,51 @@ func (c *replicaStatsMRUCache) get(extent extentID, store storeID) (val *shared.
 	}
 
 	// Check for cache hit
-	if psmap, ok = c.cache[extent]; ok {
-		if val, ok = psmap.m[store]; ok { // store == `` will fall through here
-			if val != nil {
-				return
-			}
-		}
-
-		// Get any one store's replica stats; see if it is sealed
-	oneStore:
-		for foundStore, val = range psmap.m {
-			if val != nil {
-				break oneStore
-			}
-		}
+	intrfc = c.cache.Get(string(extent))
```

Comment:
Is this cache used across destinations ? If so, same extentID across 2 destinations will stomp on each other.